README.md

```diff
@@ -185,3 +186,82 @@ make test_unit
 # Shannon E2E test only
 make test_e2e_shannon_relay
 ```
+
+## User Data
+
+By default, PATH does not associate user data with service requests.
+
+You may opt to enable user data config to unlock the ability to associate a user with a service request.
+
+This is required for:
+
+- User specified app settings
+- Metering and billing of service requests
+- Rate limiting of service requests by throughput and/or capacity
+
+### Updated Endpoint
+
+Enabling user data will modify the endpoint for service requests to require a user app ID identifier at the end of the URL path.
+
+For example:
+
+```bash
+http://eth-mainnet.localhost:3000/v1/{user_app_id}
```

Comment:
Why do we call it a user_app instead of a gateway_app?

Reply:
A few thoughts on this:

1. An application in this context relates specifically to an end user (ie. someone using an endpoint, modifying settings, getting billed for relays, etc.) so I think the naming is appropriate.

2. Given the naming of the actors in Shannon I think having Gateways, Applications AND Gateway Applications could be confusing to most people.

That said, I typed this I had the thought that perhaps `Endpoint` could be a better name for this? WDYT?

Reply:
Resolving this comment and will follow up in a new set of comments after review round #2.